server/pbench/bin/pbench-index-text-file

```diff
@@ -0,0 +1,86 @@
+#! /bin/bash
+
+prog=$(basename $0)
+
+function usage() {
+    echo "Usage: $prog --name <name> --timestamp <ts> --type status|error <file-to-index>"
+}
+
+opts=$(getopt -q -o hn:t:T: --longoptions "help,name:,timestamp:,type:" -n "getopt.sh" -- "$@");
+
+eval set -- "$opts"
+while true; do
+    case "$1" in
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        -n|--name)
+            name="$2"
+            shift 2
+            ;;
+        -t|--timestamp)
+            timestamp="$2"
+            shift 2
+            ;;
+        -T|--type)
+            doctype="$2"
+            shift 2
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+done
+
+file=$1
+
+# if no name, timestamp, type, or file bail
+if [ -z "$name" ] ;then
+    usage
+    exit 1
+fi
+
+if [ -z "$timestamp" ] ;then
+    usage
+    exit 2
+fi
+
+if [ -z "$doctype" ] ;then
+    usage
+    exit 3
+fi
+
+if [ -z "$file" ] ;then
+    usage
+    exit 1
+fi
+
+CONFIG=/opt/pbench-server/lib/config/pbench-index.cfg
+
+server=$(getconf.py server Server)
+index=$(getconf.py index_prefix Settings)-"$name"-"$timestamp"
+md5=$(md5sum $file | cut -d' ' -f1)
+
+text=$(cat $file | tr '\n' ' ')

```

Comment:
Add comment:

Translate newlines to spaces - otherwise the sed below gets very confused. It would be ideal if we could send the textfile as an attachment. Recent ES versions have a plugin for that, but the production ES is old and does not support it. We'll revisit this when we update ES.